src/bert_model.rs

```diff
@@ -192,6 +193,54 @@ impl LoadFromHDF5 for BertEmbeddings {
     }
 }
 
+#[derive(Debug)]
+pub struct BertEncoder {
+    layers: Vec<BertLayer>,
+}
+
+impl BertEncoder {
+    /// Apply the encoder.
+    ///
+    /// Returns the hidden states and attention per layer.
+    pub fn forward_t(&self, hidden_states: &Tensor, train: bool) -> Vec<(Tensor, Tensor)> {
```

Comment:
See below, maybe return a `Vec<BertLayerOutput>`?

Reply:
Good idea! I'll do that in the next PR.